Hawkular 1259

jmazzitelli opened the pull request.

jmazzitelli committed:
rip out all mvn modules that are obsolete
move to 2.0.0.Alpha1-SNAPSHOT version
this changes nothing in the agent, still basically the 1.0 agent but lots of maven stuff removed

jmazzitelli committed:
remove unused dep

jmazzitelli committed:
start using the new inventory API/model

jmazzitelli committed:
since we won't be collecting metrics, no such thing as updating collection interval.
so remove the command.
I renamed/reimplemented it as a dummy echo command because I don't want to forget the touch points when new commands are to be added, becuase I suspect new commands will be needed in teh future.
so rather than forget, I just have a echo command that we can copy later.

jmazzitelli committed:
re-introduce the inventory-context setting.
this points to the new inventory REST endpoint

jmazzitelli committed:
the start of inventory storage

jotak committed:
Inventory: build modified resources for import (#355)

* Inventory: build modified resources for import

Here all resources under a modified root are imported. Future improvement could be to import only created or modified children.

* Import only modified resources rather than whole root resource tree

jmazzitelli committed:
store units (#358)

* store units

* add log message on exception

jmazzitelli committed:
fix tests (#359)

* fix a test - other tests still fail

* tests pass now (unit tests only, not itests)

jmazzitelli committed:
put feed id in resource ids, but not resource type ids (resource type ids are back to the way they were, which is just the type name) (#361)

jmazzitelli committed:
wait for inventory to be started

jmazzitelli committed:
we need feed ID in the notification

lucasponce committed:
Preliminar migration of itests into new inventory

jmazzitelli committed:
update readme

lucasponce committed:
Adapt CmdGw itests to new Inventory API

jmazzitelli committed:
support the case when no resource type set names are specified in managed server definitions. (#366)

the default is to include all enabled default sets.

jotak committed:
Updated inventory model package

jotak committed:
Set commons dependency on srcdeps

jotak committed:
Delete several resources at once

jmazzitelli committed:
fix test

jmazzitelli committed:
use the correct name

jmazzitelli committed:
fix copy-n-paste error
after this, all itests seem to be passing now

jotak committed:
Updated inventory model

related to https://github.com/hawkular/hawkular-commons/pull/154

jmazzitelli committed:
store resource config (#372)

* store resource config

* put back in the itests

jmazzitelli committed:
temp fix to support h-metrics

jmazzitelli committed:
add metric ID

josejulio committed:
Removes type of resource from its name

mwringe committed:
HAWKULAR-1280: update the agent's docker/openshift configuration

jmazzitelli committed:
push image to docker hub

jmazzitelli committed:
opps. fix typo

jotak committed:
Update commons srcdeps

mwringe committed:
HAWKULAR-1283: Fix problem access local DRM connection in OpenShift

jshaughn commented:
Adding DNM until we have a green light on the merge/branching plan. Also, as in services, we may want to make a release of commons and then update this to use the release dep, not a src-dep.

jmazzitelli commented:
I still don't understand that failure in travis. However, ignore it. Because the next set of changes we introduce (removing h-metrics) will mean this test should again pass.

josejulio committed:
HAWKULAR-1308 - Loading sensitive information as envs instead of system properties

josejulio committed:
HAWKULAR-1289 - Build a single docker image that supports both standalone and domain

 By setting env HAWKULAR_MODE one can run it in standalone or domain.
 If HAWKULAR_MODE is not set, it defaults to standalone.
   docker run -e HAWKULAR_MODE=standalone hawkular/wildfly-hawkular-javaagent
   docker run -e HAWKULAR_MODE=domain hawkular/wildfly-hawkular-javaagent
